demo/demo_args.c

```diff
@@ -1125,6 +1131,36 @@ int demo_parse_argv(cmdline_args_t *parsed_args, int argc, char *argv[]) {
         case 305:
             parsed_args->alternative_logger = true;
             break;
+        case 306: {
+            const size_t identity_length = strlen(optarg);
+            if (parsed_args->connection_args.public_cert_or_psk_identity != NULL
+                    || identity_length == 0) {
+                demo_log(ERROR, "Invalid identity either identity was set "
+                                "twice or empty parameter was passed");
+                goto finish;
+            }
+            clone_buffer(
+                    &parsed_args->connection_args.public_cert_or_psk_identity,
+                    &parsed_args->connection_args
+                             .public_cert_or_psk_identity_size,
+                    optarg, identity_length);
```

Comment:
Here and in the usage below: you don't check the return value of `clone_buffer()`, so in the (unlikely) case of an out-of-memory condition, the user won't even get an error message.